tests/subsys/net/lib/fota_download/src/test_fota_download.c

```diff
@@ -40,9 +40,9 @@ int dfu_target_init(int img_type, int img_num, size_t file_size, dfu_target_call
 	return 0;
 }
 
-int dfu_target_img_type(const void *const buf, size_t len)
+enum dfu_target_image_type dfu_target_img_type(const void *const buf, size_t len)
 {
-	return 0;
+	return DFU_TARGET_IMAGE_TYPE_MCUBOOT;
```

Comment:
Nothing wrong here, but you also need to update the tests in tests/subsys/dfu/dfu_target/mcuboot/src/main.c as they are checking if the return value from dfu_target_image_type is > 0, which they will always be now, so the test has lost its value.

Reply:
```c

	identify_retval = true;
	ret = dfu_target_img_type(buf, sizeof(buf));
	zassert_true(ret > 0, "Valid type not recognized");
```
That test is still valid.
IMAGE_TYPE_MCUBOOT is still at the same value `1` like it used to be. Test is still correct.
Although it could verify that it is returning exactly the same value as `DFU_TARGET_IMAGE_TYPE_MCUBOOT` but still it is not zero.

I did not spot any test that had `identify_retval = false`  but that would be so trivial that it does not matter.